ros2topic/ros2topic/verb/bw.py

```diff
@@ -79,26 +80,33 @@ def add_arguments(self, parser, cli_name):
             '--window', '-w', type=positive_int, default=DEFAULT_WINDOW_SIZE,
             help='maximum window size, in # of messages, for calculating rate '
                  f'(default: {DEFAULT_WINDOW_SIZE})', metavar='WINDOW')
+        add_direct_node_arguments(parser)
 
     def main(self, *, args):
         with DirectNode(args) as node:
-            _rostopic_bw(node.node, args.topic, window_size=args.window)
+            _rostopic_bw(node.node, args.topic, window_size=args.window,
+                         use_sim_time=args.use_sim_time)
 
 
 class ROSTopicBandwidth(object):
 
-    def __init__(self, node, window_size):
+    def __init__(self, node, window_size, use_sim_time=False):
         self.lock = threading.Lock()
         self.last_printed_tn = 0
         self.sizes = []
         self.times = []
         self.window_size = window_size
+        self._clock = node.get_clock()
+        self.use_sim_time = use_sim_time
 
     def callback(self, data):
         """Execute ros sub callback."""
         with self.lock:
             try:
-                t = time.monotonic()
+                if self.use_sim_time:
+                    t = self._clock.now().nanoseconds * 1.e-9
+                else:
+                    t = time.monotonic()
```

Comment:
TODO. using `Clock(clock_type=ClockType.STEADY_TIME)` instead might be better.

Reply:
good eye, I think this should be addressed.

Reply:
updated.